torch/_dynamo/variables/distributed.py

```diff
@@ -10,10 +10,18 @@
 
 
 class DistributedVariable(VariableTracker):
-    def __init__(self, **kwargs):
+    """
+    The base distributed variable that checks for the distributed existance
+    and hold the value for a distributed object.
+    """
+    def __init__(self, value, **kwargs):
         super().__init__(**kwargs)
         if not DistributedVariable.is_available():
             unimplemented("torch.distributed package is not available!")
+        self.value = value
+
+    def python_type(self):
```

Comment:
I think this is ok.  I would like to have a docstring on the base class Variable (where `python_type` is unimplemented) which explains under what assumptions you are allowed to implement it (or shouldn't).  Otherwise, i just look at all the other variable classes, determine this looks about the same, and ship it

cc @voznesenskym

Reply:
updated the base VariableTracker with a docstr, I don't actually know when we should not implement `python_type` so only added a comment about when we should.